exporter/awsemfexporter/emf_exporter.go

```diff
@@ -56,66 +51,40 @@ type emfExporter struct {
 	collectorID   string
 }
 
-// newEmfPusher func creates an EMF Exporter instance with data push callback func
-func newEmfPusher(
-	config component.Config,
-	params exporter.CreateSettings,
-) (*emfExporter, error) {
+// newEmfExporter creates a new exporter using exporterhelper
+func newEmfExporter(config *Config, set exporter.CreateSettings) (*emfExporter, error) {
 	if config == nil {
 		return nil, errors.New("emf exporter config is nil")
 	}
 
-	logger := params.Logger
-	expConfig := config.(*Config)
-	expConfig.logger = logger
+	config.logger = set.Logger
 
 	// create AWS session
-	awsConfig, session, err := awsutil.GetAWSConfigSession(logger, &awsutil.Conn{}, &expConfig.AWSSessionSettings)
+	awsConfig, session, err := awsutil.GetAWSConfigSession(set.Logger, &awsutil.Conn{}, &config.AWSSessionSettings)
 	if err != nil {
 		return nil, err
 	}
 
 	// create CWLogs client with aws session config
-	svcStructuredLog := cwlogs.NewClient(logger, awsConfig, params.BuildInfo, expConfig.LogGroupName, expConfig.LogRetention, session)
-	collectorIdentifier, _ := uuid.NewRandom()
+	svcStructuredLog := cwlogs.NewClient(set.Logger, awsConfig, set.BuildInfo, config.LogGroupName, config.LogRetention, session)
+	collectorIdentifier, err := uuid.NewRandom()
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
why are we bothering to handle this error?

Reply:
which design principle says to accept unhandled errors because they are unlikely?

Reply:
I am more trying to get at what changed.  This is one of those type of methods that will never fail unless something is completely broken

Reply:
If its failed, it would fail at `unit test`. As billy said, why would we ignore unhandled errors, if it is in test environment ,yes I agree it should be more less robust but I don't think we want to enforce it with prod env.